src/abstract-interpretation/processor.ts

```diff
@@ -1,7 +1,157 @@
-import { DataflowInformation } from '../dataflow/internal/info'
-import { NormalizedAst } from '../r-bridge'
+import {DataflowInformation} from '../dataflow/internal/info'
+import {NodeId, NormalizedAst, ParentInformation, RBinaryOp, RType} from '../r-bridge'
+import {CfgVertexType, extractCFG} from '../util/cfg/cfg'
+import {visitCfg} from '../util/cfg/visitor'
+import {guard} from '../util/assert'
+import {DataflowGraphVertexInfo, EdgeType, OutgoingEdges} from '../dataflow'
+
+const constraintMap = new Map<NodeId, Constraints>()
+
+class Stack<ElementType> {
+	private backingStore: ElementType[] = []
+
+	size(): number { return this.backingStore.length }
+	peek(): ElementType | undefined { return this.backingStore[this.size() - 1] }
+	pop(): ElementType | undefined { return this.backingStore.pop() }
+	push(item: ElementType): ElementType {
+		this.backingStore.push(item)
+		return item
+	}
+}
+
+interface IHandler<ValueType> {
+	name:  string,
+	enter: () => void
+	exit:  () => ValueType
+	next:  (value: ValueType) => void
+}
+
+type Intervals = Set<{min: number, minInclusive: boolean, max: number, maxInclusive: boolean}>
+
+interface Constraints {
+	node:      NodeId,
+	intervals: Intervals,
+	debugMsg:  string
+}
+
+function intervalFromNumber(n: number): Intervals {
+	return new Set([{min: n, minInclusive: true, max: n, maxInclusive: true}])
+}
+
+function getIntervalsOfDfgChild(node: NodeId, dfg: DataflowInformation): Intervals {
+	const dfgNode: [DataflowGraphVertexInfo, OutgoingEdges] | undefined = dfg.graph.get(node)
+	guard(dfgNode !== undefined, `No DFG-Node found with ID ${node}`)
+	const [_, children] = dfgNode
+	const ids = Array.from(children.entries())
+		.filter(([_, edge]) => edge.types.has(EdgeType.Reads))
+		.map(([id, _]) => id)
+	if(ids.length === 0) {
+		return new Set()
+	}
+	guard(ids.length === 1, `Multiple reads-edges found for node ${node}`)
```

Comment:
But this is definitely possible, consider:
```r
if(runif(1)) x <- 1; else x <- 2; 
print(x)
```

This should consist of the union of all domains (for now, i keep that name even if we later disagree).

Reply:
Yeah, I'm aware that this can happen. I just wanted to ignore it for this commit. But a "TODO" would've been better than guarding against this case.